appcues/src/main/java/com/appcues/ui/composables/CompositionLocals.kt

```diff
@@ -94,8 +95,18 @@ internal sealed class StackScope(private val childrenCount: Int) {
     class ColumnStackScope(val width: Double?, childrenCount: Int) : StackScope(childrenCount)
 }
 
+/**
+ * Use LocalAppcuesStepMetadata to access the shared information between steps
+ *
+ * Example usage:
+ * @sample com.appcues.trait.appcues.BackdropKeyholeTrait
+ */
 val LocalAppcuesStepMetadata = compositionLocalOf<AppcuesStepMetadata> { noLocalProvidedFor("LocalAppcuesStepMetadata") }
 
+/**
+ * AppcuesStepMetadata used to share trait information between traits, generated from produceMetadata.
+ * Previous and actual refers to the steps inside of the step group
```

Comment:
Wondering if this could be more clear? `previous` refers to the previously known metadata values that could be used for animating from a previous state, is that correct? And `actual` - wondering if it should be renamed `current`? That is the current latest known metadata values, correct?

Theoretically, it is not tied to the step though right? Meaning, a single step might have the ability to have metadata updated multiple times. I know this is not really how it is used right now, its more about step transitions, but in theory, the metadata can update and publish at any time - say for instance orientation changes, or some timed behaviors

Reply:
maybe the wording is not precise indeed, but the metadata is tied to the composition, it refreshes on step change, but will also re-create, update on orientation change like you said its just not something controlled by the trait that is implementing MetadataTrait.